### PR TITLE
Migrate from CircleCI to Github Actions

### PR DESCRIPTION
**Pull request by spirosoik**
#### Summary
        The PR build (CI) is a reusable [required workflow](https://github.com/mattermost/actions-workflows/tree/main/.github/workflows/required_workflows/plugins)  

        #### Ticket Link
        Ticket: https://mattermost.atlassian.net/browse/CLD-4816

### CODE REVIEW

**Review by m1lt0n — APPROVED** [review]

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    tags:
+      - "^v.*"
+
+permissions:
+  contents: read
+
+jobs:
+  plugin-cd:
+    uses: mattermost/actions-workflows/.github/workflows/plugin-cd.yml@main
+    secrets: inherit
```
Comment (hanzei) [inline .github/workflows/cd.yml:18]: The cd workflow should not be in the starter template. This repo is community focused and the CD pipeline is Mattermost specific.

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  push:
+    tags:
+      - "^v.*"
+
+permissions:
+  contents: read
+
+jobs:
+  plugin-cd:
+    uses: mattermost/actions-workflows/.github/workflows/plugin-cd.yml@main
```
Comment (hanzei) [inline .github/workflows/cd.yml:17]: I think we should fixate the version here so community member using this template have to opt into updates.

File: .github/workflows/cd.yml (hunk 1 above)
Comment (spirosoik) [reply .github/workflows/cd.yml:18]: good let me remove this.

File: .github/workflows/cd.yml (hunk 2 above)
Comment (spirosoik) [reply .github/workflows/cd.yml:17]: like a release tag?

Comment (hanzei) [reply .github/workflows/cd.yml:17]: Yes

**Pull request merged**